### ISSUE TITLE
[2.2.3+][CE/EE][Performance] Interception Cache compilation useless

### ISSUE THREAD

**Issue by IgorVitol** [magento/magento2#17680]
### Preconditions
1. Magento 2.2.3+ Commerce/Community

### Steps to reproduce
1. Install clean Magento 2.2.3+ Community/Commerce with sample data
2. Add profiler start/stop for next methods:
`\Magento\Framework\Interception\Config\Config::initialize`
`\Magento\Framework\Interception\PluginList\PluginList::_loadScopedData`
`\Magento\Framework\ObjectManager\Config\Reader\Dom::read`
**Note:** Profiler start/stop can be added like this:
```
\Magento\Framework\Profiler::start(__METHOD__);
//Method code here
\Magento\Framework\Profiler::stop(__METHOD__);
```
3. Enable CSV **profiler** - execute "bin/magento dev:profiler:enable csvfile"
4. Enable PROD mode - execute "bin/magento deploy:mode:set production". This will run **setup:di:compile** + **static:content:deploy**
5. Note step # 7 of **di:compile** - "**Interception cache generation**".
6. Run **cache:flush**
7. Try to open any URL of your store.
8. Inspect profiler results in "**var/log/profiler.csv**".

### Expected result
1. Our custom profiler records should be executed in less than second together or not executed at all:
    - Config::initialize
    - PluginList::_loadScopedData
    - Dom::read
2. Interception config should be compiled and stored to filesystem like other compiled configs in "**/generated/metadata**" folder. E.g. by using next calls:
`\Magento\Setup\Module\Di\Compiler\Config\Writer\Filesystem::write`
`\Magento\Framework\App\ObjectManager\ConfigLoader\Compiled::load`

### Actual result
1. Profiled methods above consume about 70% of first page load after **cache:flush** (via CLI or from admin pannel). E.g. page takes 20 seconds to load. 14 sec is used by 3 above methods.
2. During **di:compile**, "**Interception cache**" compiled in this class:
`Magento\Setup\Module\Di\App\Task\Operation\InterceptionCache`
By calling:
`\Magento\Framework\Interception\Config\Config::initialize`
Which actually only store compiled config to cache storage. Which can be flushed by admins via admin panel.

Current "**Interception cache compilation**" is completely useles for PROD envs, since **di:compile** should never be launched on live web-site. Usually compiled data is prepared on another machine to minimize downtime. So, cache storage after release is not populated with such config on PROD env and this anyway takes big load time first time.
Additionally admins could flush the cache, so interception config will be regenerated once again.

**Comment by magento-engcom-team** [comment]
Hi @IgorVitol. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@IgorVitol do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by ghost** [comment]
@IgorVitol, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by magento-engcom-team** [comment]
Hi @IgorVitol. Thank you for your report. 
The issue has been fixed in magento/magento2#18648 by @pmclain in 2.3-develop branch
Related commit(s):
 - [86a30b6295fec501b0ecaa27e173bcdefc8fa95b](https://github.com/magento/magento2/commit/86a30b6295fec501b0ecaa27e173bcdefc8fa95b)
 - [32191691b5e66da54c422b4968590b7c581c0e58](https://github.com/magento/magento2/commit/32191691b5e66da54c422b4968590b7c581c0e58)
 - [a901ad3fd5390c455409d1b8e09af8190debd26c](https://github.com/magento/magento2/commit/a901ad3fd5390c455409d1b8e09af8190debd26c)
 - [7dae4c185b01b621cd51841ecb67521d8b745298](https://github.com/magento/magento2/commit/7dae4c185b01b621cd51841ecb67521d8b745298)
 - [cbf828a00a9a2287d7c7bb640f3c1078410b4de4](https://github.com/magento/magento2/commit/cbf828a00a9a2287d7c7bb640f3c1078410b4de4)
 - [4c495f2faa29edcaeebfe1ca1691754f3ef41b55](https://github.com/magento/magento2/commit/4c495f2faa29edcaeebfe1ca1691754f3ef41b55)
 - [1862d2c02744a33a5b8b1a44eab53adf1bb79048](https://github.com/magento/magento2/commit/1862d2c02744a33a5b8b1a44eab53adf1bb79048)
 - [2f5b730fa800c528e8f38da91154089f094090dc](https://github.com/magento/magento2/commit/2f5b730fa800c528e8f38da91154089f094090dc)
 - [9135654c9f123ca8a22f0aa23677fa8222912e9a](https://github.com/magento/magento2/commit/9135654c9f123ca8a22f0aa23677fa8222912e9a)
 - [5db83d30c9ca29a0f6613932196c9b0f3bc89de8](https://github.com/magento/magento2/commit/5db83d30c9ca29a0f6613932196c9b0f3bc89de8)
 - [40c58fc482c87a0840e2b1d05c3f951c30f140f2](https://github.com/magento/magento2/commit/40c58fc482c87a0840e2b1d05c3f951c30f140f2)
 - [be71e3b26e103dbecffa1f9397b20f7e794e36f3](https://github.com/magento/magento2/commit/be71e3b26e103dbecffa1f9397b20f7e794e36f3)
 - [b80d92d59882c4f686aa8e538ad167ef4630be43](https://github.com/magento/magento2/commit/b80d92d59882c4f686aa8e538ad167ef4630be43)
 - [33435b7c4ac6ca2fdbaf07a393ba8d96f97de068](https://github.com/magento/magento2/commit/33435b7c4ac6ca2fdbaf07a393ba8d96f97de068)

The fix will be available with the upcoming 2.3.1 release.
